### PR TITLE
DOC: update and seperate the Series.drop and Dataframe.drop docstring

### PR DESCRIPTION
**Pull request by JennaVergeynst**
Checklist for the pandas documentation sprint (ignore this if you are doing
an unrelated PR):

- [X] PR title is "DOC: update the <your-function-or-method> docstring"
- [X] The validation script passes: `scripts/validate_docstrings.py <your-function-or-method>`
- [X] The PEP8 style check passes: `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [X] The html version looks good: `python doc/make.py --single <your-function-or-method>`
- [X] It has been proofread on language by another sprint participant

Please include the output of the validation script below between the "```" ticks:

```
Errors found:
Errors in parameters section
Parameter "columns" description should finish with "."
```

This error is caused by `.. versionadded:: 0.21.0`, which needs to stay in place.

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -3035,6 +3035,129 @@ def reindex_axis(self, labels, axis=0, method=None, level=None, copy=True,
...
+                        big 	small
+        lama 	speed 	45.0 	30.0
+                weight 	200.0 	100.0
+        cow 	speed 	30.0 	20.0
+                weight 	250.0 	150.0
+        falcon 	speed 	320.0 	250.0
+                weight 	1.0 	0.8
+        """
+        return super(DataFrame,
+                     self).drop(labels=labels, axis=axis, index=index,
```
Comment (stijnvanhoey) [inline pandas/core/frame.py]: indendation is a bit strange here. I would set the `self` on the same line

File: pandas/core/frame.py [hunk 2]
```diff
@@ -3035,6 +3035,129 @@ def reindex_axis(self, labels, axis=0, method=None, level=None, copy=True,
...
+
+        Parameters
+        ----------
+        labels : single label or list-like
+            Index or column labels to drop.
+        axis : int or axis name, default 0
+            Whether to drop labels from the index (0 / 'index') or
+            columns (1 / 'columns').
+        index : None
+            Redundant for application on Series, but index can be used instead
```
Comment (stijnvanhoey) [inline pandas/core/frame.py]: As this is now in the `data.frame` section, this is probably not redundant?

File: pandas/core/frame.py [hunk 3]
```diff
@@ -3035,6 +3035,129 @@ def reindex_axis(self, labels, axis=0, method=None, level=None, copy=True,
...
+            of labels.
+        columns : None
+            Redundant for application on Series, but index can be used instead
```
Comment (stijnvanhoey) [inline pandas/core/frame.py]: As this is now in the `data.frame` section, this is probably not redundant?

File: pandas/core/frame.py [hunk 4]
```diff
@@ -3035,6 +3035,129 @@ def reindex_axis(self, labels, axis=0, method=None, level=None, copy=True,
...
+            of labels.
+
+            .. versionadded:: 0.21.0
+        level : int or level name, optional
+            For MultiIndex.
```
Comment (stijnvanhoey) [inline pandas/core/frame.py]: Is it useful to make this more detailed? e.g. `Level for which the labels will be removed`

File: pandas/core/frame.py [hunk 5]
```diff
@@ -3035,6 +3035,129 @@ def reindex_axis(self, labels, axis=0, method=None, level=None, copy=True,
...
+        inplace : bool, default False
+            If True, do operation inplace and return None.
+        errors : {'ignore', 'raise'}, default 'raise'
+            If 'ignore', suppress error and existing labels are dropped.
+
+        Returns
+        -------
+        dropped : type of caller
```
Comment (stijnvanhoey) [inline pandas/core/frame.py]: In the generic, this was the same as the caller, but here we know it is a dataframe

File: pandas/core/frame.py [hunk 6]
```diff
@@ -3035,6 +3035,129 @@ def reindex_axis(self, labels, axis=0, method=None, level=None, copy=True,
...
+                weight  200.0 	100.0
+                length 	1.5 	1.0
+        cow 	speed 	30.0 	20.0
+                weight 	250.0 	150.0
+                length 	1.5 	0.8
+        falcon 	speed 	320.0 	250.0
+                weight 	1.0 	0.8
+                length 	0.3 	0.2
+
+        >>> df.drop(index='cow',columns='small')
```
Comment (stijnvanhoey) [inline pandas/core/frame.py]: missing space after `, `

File: pandas/core/frame.py [hunk 7]
```diff
@@ -3035,6 +3035,129 @@ def reindex_axis(self, labels, axis=0, method=None, level=None, copy=True,
...
+
+        Drop a row by index
+
+        >>> df.drop([0, 1])
+           A  B   C   D
+        2  8  9  10  11
+
+        Drop columns and/or rows of MultiIndex
+
+        >>> midx = pd.MultiIndex(levels=[['lama','cow','falcon'],
```
Comment (stijnvanhoey) [inline pandas/core/frame.py]: missing spaces after `,`

File: pandas/core/series.py [hunk 8]
```diff
@@ -2660,6 +2660,97 @@ def rename(self, index=None, **kwargs):
...
+        level : int or level name, optional
+            For MultiIndex.
+        inplace : bool, default False
+            If True, do operation inplace and return None.
+        errors : {'ignore', 'raise'}, default 'raise'
+            If 'ignore', suppress error and existing labels are dropped.
+
+        Returns
+        -------
+        dropped : type of caller
```
Comment (stijnvanhoey) [inline pandas/core/series.py]: pandas.Series

**Review by stijnvanhoey — CHANGES_REQUESTED** [review]

File: pandas/core/frame.py [hunk 9]
```diff
@@ -3035,6 +3035,129 @@ def reindex_axis(self, labels, axis=0, method=None, level=None, copy=True,
...
+        multi-index, labels on different levels can be removed by specifying
+        the level name or int.
+
+        Parameters
+        ----------
+        labels : single label or list-like
+            Index or column labels to drop.
+        axis : int or axis name, default 0
+            Whether to drop labels from the index (0 / 'index') or
+            columns (1 / 'columns').
```
Comment (stijnvanhoey) [inline pandas/core/frame.py]: I would write (0 or 'index') and  columns (1 or 'columns')

File: pandas/core/frame.py [hunk 10]
```diff
@@ -3035,6 +3035,129 @@ def reindex_axis(self, labels, axis=0, method=None, level=None, copy=True,
...
+        Returns
+        -------
+        dropped : type of caller
+
+        See Also
+        --------
+        DataFrame.dropna : Return DataFrame with labels on given axis omitted
+            where (all or any) data are missing
+        DataFrame.drop_duplicates : Return DataFrame with duplicate rows
+            removed, optionally only considering certain columns
```
Comment (jreback) [inline pandas/core/frame.py]: reference DataFrame.loc (first) as well as this is an indexing operations

File: pandas/core/frame.py [hunk 11]
```diff
@@ -3035,6 +3035,129 @@ def reindex_axis(self, labels, axis=0, method=None, level=None, copy=True,
...
+        1  4   7
+        2  8  11
+
+        Drop a row by index
+
+        >>> df.drop([0, 1])
+           A  B   C   D
+        2  8  9  10  11
+
+        Drop columns and/or rows of MultiIndex
```
Comment (jreback) [inline pandas/core/frame.py]: MultiIndexed DataFrame

File: pandas/core/frame.py [hunk 12]
```diff
@@ -3035,6 +3035,129 @@ def reindex_axis(self, labels, axis=0, method=None, level=None, copy=True,
...
+        Remove rows or columns by specifying label names and corresponding
+        axis, or by specifying directly index or column names. When using a
+        multi-index, labels on different levels can be removed by specifying
+        the level name or int.
+
+        Parameters
+        ----------
+        labels : single label or list-like
+            Index or column labels to drop.
+        axis : int or axis name, default 0
```
Comment (stijnvanhoey) [inline pandas/core/frame.py]: cfr. gitter discussion: `axis : {0 or 'index', 1 or 'columns'}, default 0`

File: pandas/core/frame.py [hunk 13]
```diff
@@ -3035,6 +3035,129 @@ def reindex_axis(self, labels, axis=0, method=None, level=None, copy=True,
...
+        ...                              ['speed','weight','length']],
+        ...                      labels=[[0, 0, 0, 1, 1, 1, 2, 2, 2],
+        ...                              [0, 1, 2, 0, 1, 2, 0, 1, 2]])
+        >>> df = pd.DataFrame(index=midx, columns=['big','small'],
+        ...                   data=[[45,30],[200,100],[1.5,1],[30,20],
+                                    [250,150],[1.5,0.8],[320,250],[1,0.8],
+                                    [0.3,0.2]])
+        >>> df
+                        big 	small
+        lama 	speed 	45.0 	30.0
```
Comment (jreback) [inline pandas/core/frame.py]: I don't think this is layed out  correctly, a copy-paste issue?

File: pandas/core/frame.py [hunk 14]
```diff
@@ -3035,6 +3035,129 @@ def reindex_axis(self, labels, axis=0, method=None, level=None, copy=True,
...
+        >>> df.drop([0, 1])
+           A  B   C   D
+        2  8  9  10  11
+
+        Drop columns and/or rows of MultiIndex
+
+        >>> midx = pd.MultiIndex(levels=[['lama','cow','falcon'],
+        ...                              ['speed','weight','length']],
+        ...                      labels=[[0, 0, 0, 1, 1, 1, 2, 2, 2],
+        ...                              [0, 1, 2, 0, 1, 2, 0, 1, 2]])
```
Comment (jreback) [inline pandas/core/frame.py]: I would make this example a bit simpler, maybe using integers for the value column (and order them, e.g. use ``np.arange(len(midx))`` to create (big/small)

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/frame.py (hunk 10 above)
Comment (jreback) [reply pandas/core/frame.py]: also Series.drop

**Review by jreback — CHANGES_REQUESTED** [review]

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
@JennaVergeynst Thanks for the PR!

**Pull request merged**